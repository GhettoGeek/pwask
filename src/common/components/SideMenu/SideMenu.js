import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { Drawer } from '@material-ui/core'
import List from '../List'
import Icon from '../Icon'

function TemporaryDrawer({ items }) {
  const { t } = useTranslation()
  const [isOpen, setIsOpen] = React.useState(false)

  return (
    <div>
      <Icon
        name="menu"
        color="inherit"
        aria-label={t('sideMenu.openDrawer')}
        onClick={() => setIsOpen(true)}
      />
      <Drawer open={isOpen} onClose={() => setIsOpen(false)}>
        <div
          tabIndex={0}
          role="button"
          onClick={() => setIsOpen(false)}
          onKeyDown={() => setIsOpen(false)}
        >
          <div width={250}>
            <List items={items} />
          </div>
        </div>
      </Drawer>
    </div>
  )
}

TemporaryDrawer.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    icon: PropTypes.node,
    link: PropTypes.string,
  })),
}

TemporaryDrawer.defaultProps = {
  items: [],
}

export default TemporaryDrawer
