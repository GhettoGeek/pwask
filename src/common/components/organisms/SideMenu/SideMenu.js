import React from 'react'
import PropTypes from 'prop-types'
import { Link as RouterLink } from 'react-router-dom'
import { withTranslation } from 'react-i18next'
import {
  Link, Drawer, IconButton, List, ListItem, ListItemIcon, ListItemText,
} from '@material-ui/core'
import {
  Menu as MenuIcon,
  Edit as EditIcon,
} from '@material-ui/icons'

function TemporaryDrawer({ t }) {
  const [isOpen, setIsOpen] = React.useState(false)

  return (
    <div>
      <IconButton color="inherit" aria-label={t('sideMenu.openDrawer')} onClick={() => setIsOpen(true)}>
        <MenuIcon />
      </IconButton>
      <Drawer open={isOpen} onClose={() => setIsOpen(false)}>
        <div
          tabIndex={0}
          role="button"
          onClick={() => setIsOpen(false)}
          onKeyDown={() => setIsOpen(false)}
        >
          <div width={250}>
            <List>
              <ListItem button key={t('sideMenu.item1')}>
                <ListItemIcon><EditIcon /></ListItemIcon>
                <Link component={RouterLink} to="/link1">
                  <ListItemText primary={t('sideMenu.item1')} />
                </Link>
              </ListItem>
              <ListItem button key={t('sideMenu.item2')}>
                <ListItemIcon><EditIcon /></ListItemIcon>
                <Link component={RouterLink} to="/link2">
                  <ListItemText primary={t('sideMenu.item2')} />
                </Link>
              </ListItem>
              <ListItem button key={t('sideMenu.item3')}>
                <ListItemIcon><EditIcon /></ListItemIcon>
                <Link component={RouterLink} to="/link3">
                  <ListItemText primary={t('sideMenu.item3')} />
                </Link>
              </ListItem>
            </List>
          </div>
        </div>
      </Drawer>
    </div>
  )
}

TemporaryDrawer.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation()(TemporaryDrawer)
