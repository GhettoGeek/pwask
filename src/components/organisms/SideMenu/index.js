import React from 'react'
import PropTypes from 'prop-types'
import { Link as RouterLink } from 'react-router-dom'
import { withTranslation } from 'react-i18next'
import Link from '@material-ui/core/Link'
import Drawer from '@material-ui/core/Drawer'
import MenuIcon from '@material-ui/icons/Menu'
import IconButton from '@material-ui/core/IconButton'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import EditIcon from '@material-ui/icons/Edit'

class TemporaryDrawer extends React.Component {
  state = {
    open: false,
  }

  toggleDrawer = open => () => {
    this.setState({
      open,
    })
  }

  render() {
    const { open } = this.state
    const { t } = this.props

    const sideList = (
      <div width={250}>
        <List>
          <ListItem button key={t('sideMenu.item1')}>
            <ListItemIcon><EditIcon /></ListItemIcon>
            <Link component={RouterLink} to="/link1">
              <ListItemText primary={t('sideMenu.item1')} />
            </Link>
          </ListItem>
          <ListItem button key={t('sideMenu.item2')}>
            <ListItemIcon><EditIcon /></ListItemIcon>
            <Link component={RouterLink} to="/link2">
              <ListItemText primary={t('sideMenu.item2')} />
            </Link>
          </ListItem>
          <ListItem button key={t('sideMenu.item3')}>
            <ListItemIcon><EditIcon /></ListItemIcon>
            <Link component={RouterLink} to="/link3">
              <ListItemText primary={t('sideMenu.item3')} />
            </Link>
          </ListItem>
        </List>
      </div>
    )

    return (
      <div>
        <IconButton color="inherit" aria-label={t('sideMenu.openDrawer')} onClick={this.toggleDrawer(true)}>
          <MenuIcon />
        </IconButton>
        <Drawer open={open} onClose={this.toggleDrawer(false)}>
          <div
            tabIndex={0}
            role="button"
            onClick={this.toggleDrawer(false)}
            onKeyDown={this.toggleDrawer(false)}
          >
            {sideList}
          </div>
        </Drawer>
      </div>
    )
  }
}

TemporaryDrawer.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation()(TemporaryDrawer)
