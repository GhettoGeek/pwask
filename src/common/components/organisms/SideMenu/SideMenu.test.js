import React from 'react'
import { shallow } from 'enzyme'
import SideMenu from './SideMenu'

describe('SideMenu', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<SideMenu />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
