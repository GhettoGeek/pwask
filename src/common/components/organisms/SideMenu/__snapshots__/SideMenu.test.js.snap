// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu renders correctly the component 1`] = `"<div><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"sideMenu.openDrawer\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span></button></div>"`;
